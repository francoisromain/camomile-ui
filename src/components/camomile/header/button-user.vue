<template>
  <button @click="dropdownToggle" class="btn-menubar px-m py-s full-x" :class="{ active: user.dropdown }">{{ user.name }}</button>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'camomile-button',
  computed: {
    ...mapState({
      user: state => state.camomile.user,
      visible: state => state.camomile.dropdown.visible
    })
  },
  methods: {
    dropdownToggle () {
      if (this.visible) {
        this.$store.commit('camomile/dropdown/close')
      } else {
        this.$store.commit('camomile/dropdown/open', {
          content: 'dropdownUser'
        })
      }
    }
  }
}
</script>



