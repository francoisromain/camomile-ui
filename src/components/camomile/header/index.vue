<template>
  <div class="bg-inverse color-bg header">
    <div class="container">
      <div class="blobs">
        <cml-title class="blob-1-4 mb-0" />
        <cml-infos class="blob-1-4 mb-0" />
        <cml-userbutton class="blob-icon mb-0 flex-right" v-if="logged" />
      </div>
    </div>
  </div>
</template>

<script>

import cmlTitle from './title.vue'
import cmlInfos from './infos.vue'
import cmlUserbutton from './userbutton.vue'

export default {
  name: 'camomile-header',
  components: {
    cmlTitle,
    cmlInfos,
    cmlUserbutton
  },
  computed: {
    logged () {
      return this.$store.state.camomile.logged
    }
  }
}
</script>

