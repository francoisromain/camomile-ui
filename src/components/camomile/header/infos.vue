<template>
  <h6 class="menubar-infos">{{ api }} : {{ url }}</h6>
</template>

<script>
export default {
  name: 'camomile-infos',
  computed: {
    url () {
      return this.$store.state.cml.url
    },
    api () {
      return this.$store.state.cml.config.axios ? 'axios api' : 'rp api';
    }
  }
}
</script>
