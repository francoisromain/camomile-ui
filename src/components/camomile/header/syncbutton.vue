<template>
  <button @click="sync" class="btn-menubar px-m py-s full-x"><i class="icon-24 icon-24-dot" :class="{ blink: active }" ></i></button>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'camomile-sync',
  computed: {
    active () {
      return this.$store.getters['cml/sync/active']
      // return true
    }
  },
  methods: {
    sync () {
      this.$store.dispatch('cml/sync/all')
    }
  }
}
</script>



