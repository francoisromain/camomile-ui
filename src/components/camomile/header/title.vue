<template>
  <h1 class="mb-0">{{ title }}</h1>
</template>

<script>
export default {
  name: 'camomile-title',
  computed: {
    title () {
      return this.$store.state.cml.config.title
    }
  }
}
</script>
