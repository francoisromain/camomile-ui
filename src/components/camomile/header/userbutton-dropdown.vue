<template>
  <div class="dropdown">
    <div v-if="isAdmin">
      <button @click="popupOpen({ type: 'users', id: user.id, title: 'User settings', closeBtn: true, component: popupEdit })" class="btn px-m py-s full-x">Settingsâ€¦</button>
    </div>
    <div>
      <button @click="logout" class="btn px-m py-s full-x mr home">Logout</button>
    </div>
  </div>
</template>

<script>
import { mapMutations, mapActions } from 'vuex'
import popupEdit from '../utils/popup-edit.vue'

export default {
  name: 'camomile-userbutton-dropdown',
  data () {
    return {
      popupEdit
    }
  },
  computed: {
    user () {
      return this.$store.state.cml.user
    },
    isAdmin () {
      return this.$store.state.cml.isAdmin
    }
  },
  methods: {
    close () {
      this.$store.commit('cml/dropdown/close')
    },
    logout () {
      return this.$store.dispatch('cml/user/logout')
    },
    popupOpen (config) {
      this.$store.commit('cml/popup/open', config)
      this.close()
    }
  }
}
</script>

