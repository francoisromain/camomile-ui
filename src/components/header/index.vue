<template>
  <div class="bg-inverse color-bg header">
    <div class="container">
      <div class="blobs">
        <div class="blob-1-4 mb-0">
          <cml-title></cml-title>
        </div>
        <div class="blob-1-2 mb-0">
          <div class="blobs-default">
            <div class="blob-default">
              <cml-syncbutton class="mb-0 left" v-if="isLogged"></cml-syncbutton>
            </div>
            <div class="blob-auto mb-0">  
              <cml-infos v-if="isLogged"></cml-infos>
            </div>
          </div>
        </div>
        <div class="blob mb-0 flex-right">
          <cml-userbutton v-if="isLogged"></cml-userbutton>
        </div>
      </div>
    </div>
  </div>
</template>

<script>

import cmlTitle from './title.vue'
import cmlInfos from './infos.vue'
import cmlUserbutton from './userbutton.vue'
import cmlSyncbutton from './syncbutton.vue'

export default {
  name: 'camomile-header',

  components: {
    cmlTitle,
    cmlInfos,
    cmlUserbutton,
    cmlSyncbutton
  },

  computed: {
    isLogged () {
      return this.$store.state.cml.user.isLogged
    }
  }
}
</script>

