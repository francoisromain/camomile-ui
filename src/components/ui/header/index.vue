<template>
  <div class="tablet-blobs">
    <div class="tablet-blob-1-4">
      <cml-title />
    </div>
    <div
      v-if="isLogged"
      class="tablet-blob-1-2">
      <div class="blobs">
        <div class="blob">
          <cml-sync class="mb-0" />
        </div>
        <div class="blob-auto">  
          <cml-infos />
        </div>
      </div>
    </div>
    <div
      v-if="isLogged"
      class="blob flex-right">
      <cml-userbutton />
    </div>
  </div>
</template>

<script>
import cmlTitle from './title.vue'
import cmlInfos from './infos.vue'
import cmlUserbutton from './userbutton.vue'
import cmlSync from './sync.vue'

export default {
  name: 'CamomileHeader',

  components: {
    cmlTitle,
    cmlInfos,
    cmlUserbutton,
    cmlSync
  },

  computed: {
    isLogged () {
      return this.$store.state.user.isLogged
    }
  }
};
</script>
