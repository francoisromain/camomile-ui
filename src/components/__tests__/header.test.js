import { mount, shallow, createLocalVue } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Vuex from 'vuex'
import header from '../header/index.vue'
import headerTitle from '../header/title.vue'
import headerSync from '../header/sync.vue'
import headerUserbutton from '../header/userbutton.vue'

const localVue = createLocalVue()
const renderer = createRenderer()

localVue.use(Vuex)

// Header
describe('header', () => {
  let store
  const state = {
    config: {
      title: 'Camomile-ui'
    }
  }

  const user = {
    namespaced: true,
    state: {
      isLogged: false,
      name: 'lu'
    }
  }

  const dropdown = {
    namespaced: true,
    state: {
      visible: false
    },
    mutations: {
      open: jest.fn(),
      close: jest.fn()
    }
  }

  const sync = {
    namespaced: true,
    actions: {
      all: jest.fn()
    }
  }

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        cml: {
          namespaced: true,
          state,
          modules: {
            user,
            dropdown,
            sync
          }
        }
      }
    })
  })

  it('renders the correct markup before login', () => {
    const headerWrapper = shallow(header, { store, localVue })
    const titleWrapper = mount(headerTitle, { store, localVue })

    expect(headerWrapper.html()).toContain(
      '<div class="bg-inverse color-bg header"><div class="container"><div class="blobs"><div class="blob-1-4 mb-0"><!----></div> <!----> <!----></div></div></div>'
    )

    expect(titleWrapper.html()).toBe('<h1 class="mb-0">Camomile-ui</h1>')
  })

  it('renders the correct markup after login', () => {
    store.state.cml.user.isLogged = true
    const wrapper = shallow(header, { store, localVue })
    expect(wrapper.html()).toContain(
      '<div class="bg-inverse color-bg header"><div class="container"><div class="blobs"><div class="blob-1-4 mb-0"><!----></div> <div class="blob-1-2 mb-0"><div class="blobs-default"><div class="blob-default"><!----></div> <div class="blob-auto mb-0"><!----></div></div></div> <div class="blob mb-0 flex-right"><!----></div></div></div></div>'
    )
  })

  it('has not changed snapshot before login', () => {
    store.state.cml.user.isLogged = false
    const wrapper = shallow(header, { store, localVue })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw err
      expect(str).toMatchSnapshot()
    })
  })

  it('has not changed snapshot after login', () => {
    const wrapper = shallow(header, { store, localVue })
    store.state.cml.user.isLogged = true
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw err
      expect(str).toMatchSnapshot()
    })
  })

  it('syncs all list when button is clicked', () => {
    const wrapper = mount(headerSync, { store, localVue })
    const button = wrapper.find('button')
    button.trigger('click')
    expect(sync.actions.all).toHaveBeenCalled()
  })

  it('opens dropdown when user clicks on user button', () => {
    const wrapper = mount(headerUserbutton, { store, localVue })
    const button = wrapper.find('button')
    button.trigger('click')
    expect(dropdown.mutations.open).toHaveBeenCalled()
  })

  it('close dropdown when user clicks again on user button', () => {
    const wrapper = mount(headerUserbutton, { store, localVue })
    const button = wrapper.find('button')
    store.state.cml.dropdown.visible = true
    button.trigger('click')
    expect(dropdown.mutations.close).toHaveBeenCalled()
  })
})
