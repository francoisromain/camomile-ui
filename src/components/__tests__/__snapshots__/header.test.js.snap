// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header has not changed snapshot after login 1`] = `
<header class="bg-inverse color-bg header">
  <div class="container">
    <div class="blobs">
      <div class="blob-1-4 mb-0">
        <!---->
      </div>
      <div class="blob-1-2 mb-0">
        <div class="blobs-default">
          <div class="blob-default">
            <!---->
          </div>
          <div class="blob-auto mb-0">
            <!---->
          </div>
        </div>
      </div>
      <div class="blob mb-0 flex-right">
        <!---->
      </div>
    </div>
  </div>
</header>
`;

exports[`header has not changed snapshot before login 1`] = `
<header class="bg-inverse color-bg header">
  <div class="container">
    <div class="blobs">
      <div class="blob-1-4 mb-0">
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</header>
`;
